File: src/Model/FileStatement/FileTable.java
package Model.FileStatement;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

public class FileTable<K, V> implements IFileTable<K, V> {

    private Map<K,V> map = new HashMap<>();

    @Override
    public void add(K key, V value) {
        map.put(key,value);
    }

    @Override
    public void delete(K key) {
        map.remove(key);
    }

    @Override
    public V get(K key) {
        return map.get(key);
    }

    @Override
    public boolean contains(K key) {
        return map.containsKey(key);
    }

    @Override
    public Iterable<K> getAllKeys() {
        return map.keySet();
    }

    @Override
    public Collection<V> getAllValues() {
        return map.values();
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        for(Map.Entry<K,V> entry : map.entrySet()){
            builder.append('<').append(entry.getKey()).append(',').append(entry.getValue()).append(">;");
        }
        return builder.toString();
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Containers/FileTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model.CustomException;

namespace Toy_Language_CS_Interpreter.Model.Containers
{
    class FileTable<K, V> : IFileTable<K, V>
    {
        private Dictionary<K, V> dictionary;

        public FileTable()
        {
            dictionary = new Dictionary<K, V>();
        }

        public void Add(K key, V value)
        {
            dictionary[key] = value;
        }

        public bool Contains(K key)
        {
            return dictionary.ContainsKey(key);
        }

        public void Delete(K key)
        {
            dictionary.Remove(key);
        }

        public V Get(K key)
        {
            if (dictionary.ContainsKey(key))
                return this.dictionary[key];
            else
                throw new NullEntryException("Key not exist in table");
        }

        public List<K> GetAllKeys()
        {
            return new List<K>(dictionary.Keys);
        }

        public List<V> GetAllValues()
        {
            return new List<V>(dictionary.Values);
        }
        public override string ToString()
        {
            StringBuilder stringBuilder = new StringBuilder();
            GetAllKeys().ForEach(e => stringBuilder.Append(e + "-->" + Get(e)).Append("\r\n"));
            return stringBuilder.ToString();
        }
    }
}
